frontend/src/pages/auth/signup/SignUp.tsx

```diff
@@ -1,12 +1,156 @@
-import React from 'react';
+import React, {useEffect, useState} from 'react';
+import {Box, Button, Checkbox, Link, Radio, Typography} from '@mui/material';
+import {Paths} from "../../../common/constansts/paths";
 import {Titles} from "../../../common/constansts/titles";
+import "../../../common/styles/button.css";
+import "./styles.css";
+import "../styles.css";
+import {useNavigate} from "react-router-dom";
+import {Buttons} from "../../../common/constansts/buttons";
+import {AuthContent} from "../../../common/constansts/authContent";
+import {Roles} from "../../../common/constansts/roles";
+import {signUpService} from "../../../services/user";
+import AuthLogo from "../components/logo/AuthLogo";
+import EmailField from "../components/fields/email/EmailField";
+import NameField from "../components/fields/name/NameField";
+import PasswordField from "../components/fields/password/PasswordField";
+import {useAppDispatch} from "../../../stores/redux/hooks";
+import {createNotification} from "../../../components/notification/Notification";
+import {Notifications} from "../../../common/constansts/notifications";
+import {showNotification} from "../../../stores/redux/notification/notificationSlice";
+import {FieldsValidation} from "./types";
+import {UserCredentials, UserType} from "../../../types/user";
+import {NotificationType} from "../../../types/notification";
+
+const initialUserCredentials: UserCredentials = {
+    email: "",
+    name: "",
+    password: ""
+}
+
+const initialFieldsValidation: FieldsValidation = {
+    isEmailValid: false,
+    isPasswordValid: false,
+    isNameValid: false
+};
 
 const SignUp: React.FC = () => {
+    const [userCredentials, setUserCredentials] = useState<UserCredentials>(initialUserCredentials);
+    const [selectedRole, setSelectedRole] = useState<UserType>(UserType.Customer);
+    const [agreeChecked, setAgreeChecked] = useState(false);
+    const [fieldsValidation, setFieldsValidation] = useState<FieldsValidation>(initialFieldsValidation);
+    const [isValid, setIsValid] = useState(false);
+    const navigate = useNavigate();
+    const dispatch = useAppDispatch();
+
+    const handleRoleChange = (event: React.ChangeEvent<HTMLInputElement>) => setSelectedRole(event.target.value as UserType);
+    const handleAgreeCheckboxChange = (event: React.ChangeEvent<HTMLInputElement>) => setAgreeChecked(event.target.checked);
+
+    const handleSignUp = async () => {
+        if (isValid) {
+            await signUpService(userCredentials, selectedRole);
+            navigate(Paths.SignIn);
+            const notification = createNotification(Notifications.SIGN_UP, NotificationType.SUCCESS);
+            dispatch(showNotification(notification));
+        } else {
+            const notification = createNotification(Notifications.VALIDATION_NOT_PASSED, NotificationType.ERROR);
```

Comment:
are you sure that the only error is VALIDATION_NOT_PASSED during registration? Maybe email already exists or server is down